Remotion/Validation/UnitTests/Rules/PropertyValidationRuleTest.cs

```diff
@@ -1,33 +1,242 @@
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, null, new[] { validatorMock.Object });
+
+      var context = new ValidationContext(null);
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.True);
+    }
+
+    [Test]
+    public void IsActive_WithCondition_WithoutInstanceToValidate_ReturnsFalse ()
+    {
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      Func<object, object> propertyFunc = obj => 42;
+      Func<object, bool> conditionFunc = obj => false;
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, Array.Empty<IPropertyValidator>());
+
+      var context = new ValidationContext(null);
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.False);
+    }
+
+
+    [Test]
+    public void IsActive_ReturnsTrueFromCondition ()
+    {
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      Func<object, object> propertyFunc = obj => obj;
+      Func<object, bool> conditionFunc = _ => true;
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, Array.Empty<IPropertyValidator>());
+
+      var context = new ValidationContext(new Person());
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.True);
+    }
+
+    [Test]
+    public void IsActive_ReturnsFalseFromCondition ()
+    {
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      Func<object, object> propertyFunc = obj => obj;
+      Func<object, bool> conditionFunc = _ => false;
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, Array.Empty<IPropertyValidator>());
+
+      var context = new ValidationContext(new Person());
+      var result = rule.IsActive(context);
+
+      Assert.That(result, Is.False);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void ToString_Overridden ()
     {
-      PropertyValidationRule<Customer, string> validationRule = null;
+      var declaringTypeMock = new Mock<ITypeInformation>();
+      declaringTypeMock
+          .Setup(d => d.FullName)
+          .Returns("Remotion.Validation.IntegrationTests.TestDomain.ComponentA.Address");
+
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+      propertyInformationMock
+          .Setup(s => s.Name)
+          .Returns("PostalCode");
+
+      propertyInformationMock
+          .Setup(p => p.DeclaringType)
+          .Returns(declaringTypeMock.Object);
+
+      Func<object, object> propertyFunc = obj => obj;
+      Func<object, bool> conditionFunc = _ => false;
+
+      PropertyValidationRule<Customer, string> validationRule =
+          new PropertyValidationRule<Customer, string>(propertyInformationMock.Object, propertyFunc, conditionFunc, Array.Empty<IPropertyValidator>());
```

Comment:
```suggestion
      PropertyValidationRule<Customer, string> validationRule =
          new PropertyValidationRule<Customer, string>(propertyInformationMock.Object, obj => null, null, new IPropertyValidator[0]);
```
I recommend dropping unsed code.

And the array syntax: passing an empty array instead of using Array.Empty makes the code idomatically easier to read. It may be personal preference or simply due to the fact that empty arrays have been arround longer than Array.Empty so there's more pattern recognition, though. To me it simply pops more when I see the empty array during the review than Array.Empty. It's different in prod-code. There, the Array.Empty is valuable because it saves a memory allocation.